starlite/parsers.py

```diff
@@ -78,3 +79,19 @@ def parse_form_data(media_type: "RequestEncodingType", form_data: "FormMultiDict
         if field.shape is SHAPE_SINGLETON and field.type_ in (UploadFile, MultipartUploadFile) and values_dict:
             return list(values_dict.values())[0]
     return values_dict
+
+
+def parse_cookie_string(cookie_string: str) -> Dict[str, str]:
+    """
+    Parses a cookie string into a dictionary of values.
+    Args:
+        cookie_string: A cookie string.
+
+    Returns:
+        A string keyed dictionary of values
+    """
+    output: Dict[str, str] = {}
+    cookies = [cookie.split("=", 1) if "=" in cookie else ("", cookie) for cookie in cookie_string.split(";")]
+    for k, v in filter(lambda x: x[0] or x[1], ((k.strip(), v.strip()) for k, v in cookies)):
+        output[k] = unquote(unquote_cookie(v))
```

Comment:
Why do we use `urllib.parse.unquote` here on top of `http.cookies._unquote`? It seems Starlette simply returns values after applying `http.cookies._unquote`  as can be seen [here](https://github.com/encode/starlette/blob/master/starlette/requests.py#L54)

Reply:
I saw, its copied verbatim from Django. But I then ran the code against a bunch of tests i created - based on tests for very popular cookie parsing tools in JS (40 million monthly downloads), and it failed for this value:

`("email=%20%22%2c%3b%2f", {"email": ' ",;/'})``

without using the urllib unquote there, the result of the value is: `{"email": ' "%20%22%2c%3b%2f'}`